espresso/streamer.go

```diff
@@ -100,12 +103,16 @@ func (s *EspressoStreamer[B]) Reset() {
 // Handle both L1 reorgs and batcher restarts by updating our state in case it is
 // not consistent with what's on the L1. Returns true if the state was updated.
 func (s *EspressoStreamer[B]) Refresh(ctx context.Context, syncStatus *eth.SyncStatus) (bool, error) {
-	s.Log.Info("Safe L2 ", "block number", syncStatus.SafeL2.Number)
+	s.Log.Info("Refreshing streamer...")
+	s.Log.Info("L2 ", "safe block number", syncStatus.SafeL2.Number)
+	s.Log.Info("L1 ", "finalized block number", syncStatus.FinalizedL1.Number, "safe block number", syncStatus.SafeL1.Number)
+	s.finalizedL1 = syncStatus.FinalizedL1
+
+	// NOTE: be sure to update s.finalizedL1 before checking this condition and returning
```

Comment:
Is there a way to update `s.finalizedL1` upon initialization, so that we don't have to set it each time again in `Refresh`?

(No need to address this if it's too complicated, since it's not for correctness.)

Reply:
Not sure I understand. `s.finalizedL1` needs to be updated periodically so that batches in the remaining list can be processed.

Reply:
All good! Just my own confusion--I thought it's setting `L1FinalizedDistance`.